infra/security.tf

```diff
@@ -52,7 +51,28 @@ module "iam_assumable_role_with_oidc" {
 
   number_of_role_policy_arns = 1
 
-  role_policy_arns = [module.iam_policy.arn]
+  role_policy_arns = [module.iam_policy.arn, "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"]
 
   oidc_fully_qualified_audiences = ["sts.amazonaws.com"]
 }
+
+module "iam_eks_role" {
```

Comment:
You are missing "oidc_providers" part.
There you will define mapping of EKS svc account and namespace to eks oidc provider arn.